src/Diagnostics.ModelsAndUtils/DataTableUtility.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Diagnostics.ModelsAndUtils;
+using Diagnostics.ModelsAndUtils.Models;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+namespace Diagnostics.ModelsAndUtils
+{
+    public static class DataTableUtility
+    {
+        public static DataTable GetDataTable(DataTableResponseObject dataTableResponse)
+        {
+            if (dataTableResponse == null)
+            {
+                throw new ArgumentNullException("dataTableResponse");
+            }
+
+            var dataTable = new DataTable(dataTableResponse.TableName);
+
+            dataTable.Columns.AddRange(dataTableResponse.Columns.Select(column => new DataColumn(column.ColumnName, GetColumnType(column.ColumnType, column.DataType))).ToArray());
+
+            foreach (var row in dataTableResponse.Rows)
+            {
+                var rowWithCorrectTypes = new List<object>();
+                for (int i = 0; i < dataTable.Columns.Count; i++)
+                {
+                    object rowValueWithCorrectType = null;
+
+                    if (row[i] != null)
+                    {
+                        rowValueWithCorrectType = Convert.ChangeType(row[i], dataTable.Columns[i].DataType);
+                    }
+
+                    rowWithCorrectTypes.Add(rowValueWithCorrectType);
+                }
+
+                dataTable.Rows.Add(rowWithCorrectTypes.ToArray());
+            }
+
+            return dataTable;
+        }
+
+        public static DataTableResponseObject GetDataTableResponseObject(DataTable table)
+        {
+            var dataTableResponseObject = new DataTableResponseObject();
+
+            dataTableResponseObject.TableName = table.TableName;
+
+            var columns = new List<DataTableResponseColumn>();
+            foreach(DataColumn col in table.Columns)
+            {
+                columns.Add(new DataTableResponseColumn() { ColumnName = col.ColumnName, DataType = col.DataType.ToString().Replace("System.", "") });
+            }
+
+            var rows = new List<string[]>();
+            foreach(DataRow row in table.Rows)
+            {
+                rows.Add(row.ItemArray.Select(x => x.ToString()).ToArray());
+            }
+
+            dataTableResponseObject.Columns = columns;
+            dataTableResponseObject.Rows = rows.ToArray();
+
+            return dataTableResponseObject;
+        }
+
+        internal static Type GetColumnType(string type, string datatype)
+        {
+            return Type.GetType($"System.{datatype}");
```

Comment:
Is `GetType()` case-sensitive? Also check this out, I think these are all datatypes supported by kusto that could possibly return from the HTTP call. https://kusto.azurewebsites.net/docs/queryLanguage/scalar-data-types/bool.html.

I notice that one of the data types is `real` and .NET doesn't have `System.Real`.

Reply:
real maps to System.Double, so a real column returns something like this:

`
{
  columnName: "ColumnName"
  dataType: "Double"
  columnType: "real"
}
`

In this case I am looking at data type, which is System.xxxx and the column type is the kusto scalar type